plugins/colorschemes/kanagawa.nix

```diff
@@ -5,195 +5,217 @@
   config,
   ...
 }:
-with lib; let
-  cfg = config.colorschemes.kanagawa;
-in {
-  options = {
-    colorschemes.kanagawa =
-      helpers.neovim-plugin.extraOptionsOptions
-      // {
-        enable = mkEnableOption "kanagawa";
-
-        package = helpers.mkPackageOption "kanagawa" pkgs.vimPlugins.kanagawa-nvim;
-
-        compile = helpers.defaultNullOpts.mkBool false "Enable compiling the colorscheme.";
-
-        undercurl = helpers.defaultNullOpts.mkBool true "Enable undercurls.";
-
-        commentStyle =
-          helpers.defaultNullOpts.mkNullable
-          types.attrs
-          "{italic = true;}"
-          "Highlight options for comments.";
-
-        functionStyle =
-          helpers.defaultNullOpts.mkNullable
-          types.attrs
-          "{}"
-          "Highlight options for functions.";
-
-        keywordStyle =
-          helpers.defaultNullOpts.mkNullable
-          types.attrs
-          "{italic = true;}"
-          "Highlight options for keywords.";
-
-        statementStyle =
-          helpers.defaultNullOpts.mkNullable
-          types.attrs
-          "{bold = true;}"
-          "Highlight options for statements.";
-
-        typeStyle =
-          helpers.defaultNullOpts.mkNullable
-          types.attrs
-          "{}"
-          "Highlight options for types.";
-
-        transparent = helpers.defaultNullOpts.mkBool false "Whether to set a background color.";
-
-        dimInactive = helpers.defaultNullOpts.mkBool false ''
-          Whether dim inactive window `:h hl-NormalNC`.
-        '';
+with lib;
+  helpers.neovim-plugin.mkNeovimPlugin config {
+    name = "kanagawa";
+    isColorscheme = true;
+    originalName = "kanagawa.nvim";
+    defaultPackage = pkgs.vimPlugins.kanagawa-nvim;
+
+    maintainers = [maintainers.GaetanLepage];
+
+    # TODO introduced 2024-03-15: remove 2024-05-15
+    deprecateExtraOptions = true;
+    imports = let
+      basePluginPath = ["colorschemes" "kanagawa"];
+    in
+      (
+        map
+        (
+          optionName:
+            mkRenamedOptionModule
+            (basePluginPath ++ [optionName])
+            (basePluginPath ++ ["settings" optionName])
+        )
+        [
+          "compile"
+          "undercurl"
+          "commentStyle"
+          "functionStyle"
+          "keywordStyle"
+          "statementStyle"
+          "typeStyle"
+          "transparent"
+          "dimInactive"
+          "terminalColors"
+          "colors"
```

Comment:
shouldn't this be colors.palette & colors.theme ?

Reply:
If I'm not mistaking, as those are already "snake case", we can limit the warning to the root option.

Reply:
Yeah, but `colors` is not an option though? It simply is an attrset?

Reply:
I've switched to writing the full paths.

Reply:
This does not work because you need to have a `isString` or `isList`